ps_mem.py

```diff
@@ -145,72 +145,59 @@ def open(self, *args):
 #
 
 def parse_options():
-    try:
-        long_options = [
-            'split-args',
-            'help',
-            'version',
-            'total',
-            'discriminate-by-pid',
-            'swap'
-        ]
-        opts, args = getopt.getopt(sys.argv[1:], "shtdSp:w:", long_options)
-    except getopt.GetoptError:
-        sys.stderr.write(help())
-        sys.exit(3)
-
-    if len(args):
-        sys.stderr.write("Extraneous arguments: %s\n" % args)
-        sys.exit(3)
-
-    # ps_mem.py options
-    split_args = False
-    pids_to_show = None
-    discriminate_by_pid = False
-    show_swap = False
-    watch = None
-    only_total = False
-
-    for o, a in opts:
-        if o in ('-s', '--split-args'):
-            split_args = True
-        if o in ('-t', '--total'):
-            only_total = True
-        if o in ('-d', '--discriminate-by-pid'):
-            discriminate_by_pid = True
-        if o in ('-S', '--swap'):
-            show_swap = True
-        if o in ('-h', '--help'):
-            sys.stdout.write(help())
-            sys.exit(0)
-        if o in ('--version'):
-            sys.stdout.write('3.13'+'\n')
-            sys.exit(0)
-        if o in ('-p',):
-            try:
-                pids_to_show = [int(x) for x in a.split(',')]
-            except:
-                sys.stderr.write(help())
-                sys.exit(3)
-        if o in ('-w',):
-            try:
-                watch = int(a)
-            except:
+    """
+    Parses options given after program call.
+
+    Returns:
+        Tuple of:
+            split_args - type boolean,
+            pids_to_show - type list or nonetype,
+            watch - type float or nonetype,
+            total - type boolean,
+            discriminate_by_pid - type boolean,
+            swap - type boolean
+    """
+    # usage=help() to show full help message upon all errors.
+    parser = argparse.ArgumentParser(usage=help())
+    # Replace default argparse help format for calls to '-h'.
+    # Adds 'usage: ' because calls to '-h' option does not add it.
+    parser.format_help = lambda: 'usage: ' + help()
+
+    parser.add_argument('--version', action='version', version='3.13')
+    parser.add_argument('-s', '--split-args', action='store_true')
+    parser.add_argument('-t', '--total', action='store_true')
+    parser.add_argument('-d', '--discriminate-by-pid', action='store_true')
+    parser.add_argument('-S', '--swap', action='store_true')
+    parser.add_argument('-p', '--pids_to_show', type=str)
+    parser.add_argument('-w', '--watch', nargs='?', const=2, type=float)
```

Comment:
This is a bug; change `const=2` into `default=2`.
* `const` is for internal values that are not read from the command line.
* `default` is used as a default value if the user did not provide the argument on the command line.